### ISSUE TITLE
transform_fold fails when using as_ argument on v4.1

### ISSUE THREAD

**Issue by david-berthelot** [altair-viz/altair#2525]
From the code examples: https://altair-viz.github.io/gallery/scatter_with_loess.html

```python
import altair as alt
import pandas as pd
import numpy as np

np.random.seed(1)

source = pd.DataFrame({
    'x': np.arange(100),
    'A': np.random.randn(100).cumsum(),
    'B': np.random.randn(100).cumsum(),
    'C': np.random.randn(100).cumsum(),
})

base = alt.Chart(source).mark_circle(opacity=0.5).transform_fold(
    fold=['A', 'B', 'C'],
    as_=['category', 'y']
).encode(
    alt.X('x:Q'),
    alt.Y('y:Q'),
    alt.Color('category:N')
)
```

Fails with a long stack trace, the last lines being:
```
    return schema.get("$id", "")
AttributeError: 'list' object has no attribute 'get'
---

I'm running v4.1.0 on Linux and it works in the Vega-Lite editor.

**Comment by jakevdp** [comment]
Thanks for the report - this issue is a duplicate of #2496, and was fixed in #2503

The best fix would be to avoid jsonschema 4.X, which is also problematic for other reasons (see #2501)
```
pip install jsonschema<4.0
```

**Comment by david-berthelot** [comment]
Thanks for the quick response, and sorry for wasting your time, I only searched open issues - I'll try to remember next time to also search for closed issues.
